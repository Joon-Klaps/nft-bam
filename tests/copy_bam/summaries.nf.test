nextflow_process {

    name "Test Process COPY_BAM"
    script "./main.nf"
    process "COPY_BAM"

    tag "summaries"

    test("statistics") {

        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/test.paired_end.bam", checkIfExists:true),
                    file("${baseDir}/tests/test.paired_end.sam", checkIfExists:true),
                    file("${baseDir}/tests/test.paired_end.cram", checkIfExists:true)
                )
                """
            }
        }

        then {
            assert snapshot(process.out.bam.collect { bam(it, "${baseDir}/tests/genome.fasta").getStatistics() }).match("statistics")
        }

    }

}
