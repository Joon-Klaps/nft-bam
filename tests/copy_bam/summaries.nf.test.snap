{
    "statistics": {
        "content": [
            [
                {
                    "maxReadLength": 151,
                    "minReadLength": 53,
                    "meanReadLength": 69,
                    "maxQuality": 60,
                    "minQuality": 0,
                    "meanQuality": 28,
                    "readCount": 400
                },
                {
                    "maxReadLength": 151,
                    "minReadLength": 53,
                    "meanReadLength": 69,
                    "maxQuality": 60,
                    "minQuality": 0,
                    "meanQuality": 28,
                    "readCount": 400
                },
                {
                    "maxReadLength": 151,
                    "minReadLength": 53,
                    "meanReadLength": 69,
                    "maxQuality": 60,
                    "minQuality": 0,
                    "meanQuality": 28,
                    "readCount": 400
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T14:04:58.524275656"
    }
}